lang/csharp/src/apache/main/Reflect/ClassCache.cs

```diff
@@ -214,8 +215,12 @@ public void LoadClassCache(Type objType, Schema s)
                     var c = GetClass(rs);
                     foreach (var f in rs.Fields)
                     {
-                        var t = c.GetPropertyType(f);
-                        LoadClassCache(t, f.Schema);
+                        if (!_previousFields.ContainsKey(f.Name))
+                        {
+                            _previousFields.TryAdd(f.Name, f.Schema);
```

Comment:
I believe you can collapse the omit the `ContainsKey` by checking the return value of `TryAdd`. The docs for `TryAdd` [state](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.tryadd?view=net-5.0#returns) that it returns: 

> true if the key/value pair was added to the ConcurrentDictionary<TKey,TValue> successfully; false if the key already exists.

So, maybe you can do something like this:

```suggestion
                        if (_previousFields.TryAdd(f.Name, f.Schema);)
                        {
```